src/renderer/components/ModalQuizz/ModalQuizz.tsx

```diff
@@ -0,0 +1,221 @@
+   *    if defined, it will override answerExplanation
+   */
+  correctAnswerExplanation?: string;
+  /**
+   * Illustration to display on the right in case of an incorrect answer
+   */
+  IncorrectAnswerIllustration?: React.ReactNode;
+  /**
+   * generic title to display in case the user picks an incorrect answer
+   */
+  incorrectAnswerTitle: string;
+  /**
+   * generic explanation to display in case the user picks an incorrect answer
+   *    if defined, it will override answerExplanation
+   */
+  incorrectAnswerExplanation?: string;
+};
+
+export type Props = {
+  title: string;
+  StartScreen: React.ReactNode;
+  started?: boolean;
+  steps: Array<QuizzStep>;
+  onClose: () => void;
+  isOpen: boolean;
+  dismissable?: boolean;
+};
+
+const ModalQuizz: React.FunctionComponent<Props> = ({
+  title,
+  steps,
+  isOpen,
+  onClose = () => {},
+  started,
+  StartScreen,
+  dismissable = true,
+}: Props) => {
+  const [stepIndex, setStepIndex] = useState(0);
+  const stepCount = steps.length;
+
+  const { t } = useTranslation();
+
+  const [userChoiceIndex, setUserChoiceIndex] = useState(undefined);
+
+  const onClickContinue = useCallback(() => {
+    setUserChoiceIndex(undefined);
+    if (stepIndex >= stepCount - 1) {
+      // TODO: handle continue pressed on last question;
+      onClose();
+      setStepIndex(0);
+    } else {
+      setStepIndex(stepIndex + 1);
+    }
+  }, [stepIndex, stepCount, setStepIndex, setUserChoiceIndex, onClose]);
+
+  const {
+    title: stepTitle,
+    choices,
+    answerExplanation,
+    correctAnswerExplanation,
+    CorrectAnswerIllustration,
+    correctAnswerTitle,
+    Illustration,
+    incorrectAnswerExplanation,
+    IncorrectAnswerIllustration,
+    incorrectAnswerTitle,
+  } = steps[stepIndex];
+
+  const userMadeAChoice = userChoiceIndex !== undefined;
+  const userChoice = userMadeAChoice ? choices[userChoiceIndex] : undefined;
+  const isCorrectChoice = userChoice ? userChoice.correct : undefined;
+
+  const radioName = `quizz-${title}-step-${stepIndex}`;
+
+  const AsideLeft = (
+    <Flex flexDirection="column" rowGap={12}>
+      <Text variant="h5">{stepTitle}</Text>
+      <Radio
+        name={`quizz-${title}-step-${stepIndex}`}
+        onChange={value => !userMadeAChoice && setUserChoiceIndex(value)}
+        currentValue={userChoiceIndex}
+        containerProps={{ flexDirection: "column", rowGap: 5 }}
+      >
+        {choices.map(({ label, correct }: QuizzChoice, index: number) => (
+          <RadioElement
+            key={`${radioName}-choice-${index}`}
+            label={label}
+            value={`${index}`}
+            variant={userMadeAChoice ? (correct ? "success" : "error") : "default"}
+          />
+        ))}
+      </Radio>
+    </Flex>
+  );
+
+  const rightSideIllustration =
+    (userMadeAChoice
+      ? isCorrectChoice
+        ? CorrectAnswerIllustration
+        : IncorrectAnswerIllustration
+      : Illustration) || Illustration;
```

Comment:
Yes because `CorrectAnswerIllustration` and `IncorrectAnswerIllustration` are optional and if undefined it will fallback to `Illustration`, maybe this will be clearer:

```js
const rightSideIllustration = userMadeAChoice
  ? isCorrectChoice
    ? CorrectAnswerIllustration || Illustration
    : IncorrectAnswerIllustration || Illustration
  : Illustration;
```

Reply:
Ah I see, thanks for explaining! 👍